superset/db_engine_specs/hive.py

```diff
@@ -530,7 +534,23 @@ def get_function_names(cls, database: "Database") -> List[str]:
         :param database: The database to get functions for
         :return: A list of function names useable in the database
         """
-        return database.get_df("SHOW FUNCTIONS")["tab_name"].tolist()
+        df = database.get_df("SHOW FUNCTIONS")
+        if cls._show_functions_column in df:
+            return df[cls._show_functions_column].tolist()
+
+        columns = df.columns.values.tolist()
+        logger.error(
```

Comment:
When this function is called in core.py the exceptions are caught and logged there. What's the philosophy in this case? Should we catch/log early or let errors bubble up and be caught in one place?

Reply:
In this case I wanted to log here where we have more context, so that we can quickly fix the problem when we see it in the logs. Also, even if the result is not as expected we can still try our best and use the results if they have a single column.